components/onfleet/actions/create-pickup-dropoff-task/create-pickup-dropoff-task.ts

```diff
@@ -0,0 +1,369 @@
+      props = {
+        ...props,
+        dropoffRecipientName: {
+          type: "string",
+          label: "Dropoff Recipient Name",
+          description: "The person to be notified that the pickup is occurring.",
+        },
+        dropoffRecipientPhone: {
+          type: "string",
+          label: "Dropoff Recipient Phone",
+          description: "The phone of the dropoff recipient.",
+        },
+        dropoffRecipientNotes: {
+          type: "string",
+          label: "Dropoff Recipient Notes",
+          description: "Notes related the recipient or the destination like door codes, beware of dog, etc.",
+          optional: true,
+        },
+      };
+    }
+    if (this.autoAssign) {
+      if (this.autoAssign != "auto-assign") {
+        props = {
+          ...props,
+          team: {
+            type: "string",
+            label: "Team Id",
+            description: "The team's Id",
+            options: async () =>  {
+              const teams = await this.onfleet.listTeams();
+
+              return teams.map(({
+                id: value,
+                name: label,
+              }) => ({
+                label,
+                value,
+              }));
+            },
+          },
+        };
+      }
+
+      if (this.autoAssign === "driver") {
+        props = {
+          ...props,
+          worker: {
+            type: "string",
+            label: "Driver",
+            description: "The driver's Id",
+            options: async () =>  {
+              const drivers = await this.onfleet.listWorkers({
+                teamId: this.team,
+              });
+
+              return drivers.map(({
+                id: value,
+                name: label,
+              }) => ({
+                label,
+                value,
+              }));
+            },
+          },
+        };
+      }
+    }
+    return props;
+  },
+  methods: {
+    async createTask(onfleet, data) {
+      const preparedData = prepareData(data);
+      return await onfleet.createTask(clearObj(preparedData));
+    },
+  },
+  async run({ $ }) {
+    const {
+      onfleet,
+      pickupRecipientName,
+      pickupRecipientPhone,
+      pickupRecipientNotes,
+      dropoffRecipientName,
+      dropoffRecipientPhone,
+      dropoffRecipientNotes,
+      taskLinkOrder,
+      pickupLatitude,
+      pickupLongitude,
+      pickupAddress,
+      dropoffLatitude,
+      dropoffLongitude,
+      dropoffAddress,
+      recipientPickup,
+      recipientDropoff,
+      ...data
+    } = this;
+
+    let response = {};
+    let summary = "";
+    if (taskLinkOrder == "1") {
+      const pickup = await this.createTask(onfleet, {
+        ...data,
+        recipientName: pickupRecipientName,
+        recipientPhone: pickupRecipientPhone,
+        recipientNotes: pickupRecipientNotes,
+        recipient: recipientPickup,
+        latitude: pickupLatitude,
+        longitude: pickupLongitude,
+        address: pickupAddress,
```

Comment:
In fact, the first one is pickup and the second one is dropoff, there isn't duplication

Reply:
Pickup and dropoff are different from each other, yes, but I meant the part that is duplicated between the scenarios. For example, instead of:
```
if (...) {
  const pickup = await this.createTask(onfleet, {
    prop1,
    prop2,
  }
} else {
  const pickup = await this.createTask(onfleet, {
    ...data,
    prop1,
    prop2,
  }
}
```
You could have something like:
```
const pickupObj = {
  prop1,
  prop2
};
if (...) {
  const pickup = await this.createTask(onfleet, {
     ...pickupObj
  }
} else {
  const pickup = await this.createTask(onfleet, {
    ...data,
    ...pickupObj
  }
}
```